### DISCUSSION TITLE
Combine fetchInfiniteQuery with useInfiniteQuery

### DISCUSSION THREAD

**Discussion by spacepolice10** [TanStack/query#7208] [Q&A] [answered]
I can't wrap around my head how should I use `fetchInfiniteQuery` in a correct manner. Let's say I have a `loader` in my React Router that fetches infinite list powered by `queryClient`. At the same time I have a hook `useInfiniteQuery` with the same key. The problem is that these things are seem to be disconnected in some ways. For example, it always says that `hasNextPage` equals `false` and ignores `getNextPageParam` property passed to the hook (event if I use the same function inside `fetchInfiiteQuery` options). Is it possible to combine `fetchInfiniteQuery` with a hook without refetching one?

**Comment by TkDodo** [comment]
this should work, please show a minimal reproduction

**Reply by spacepolice10** [reply]
I'm so sorry. I started making reproduction & found out that I have incorrectly rewrote `inititalPageParam` with value that appears to be blocking `hasNextPage` from returning TRUE. There's no problem in Tanstack Query

**Reply by TkDodo** [reply, marked as answer]
no worries, this happens 99% of the time when doing reproductions, and it's the reason why I require a reproduction before I dig into an alleged issue :)
